common/beerocks/tlvf/yaml/beerocks/tlvf/beerocks_message_apmanager.yaml

```
@@ -27,6 +27,9 @@ cACTION_APMANAGER_JOINED_NOTIFICATION:
   _type: class
   params: sNodeHostap
   cs_params: sApChannelSwitch
+  supported_channels_list:
+    _type: beerocks::message::sWifiChannel 
+    _length: [ "beerocks::message::SUPPORTED_CHANNELS_LENGTH" ]  
```

Since it is the last member, you don't need to specify the length, it can be dynamic.
Also - no need to say its a list, the 's' in the channels is enough.
So I would change it to:
```
supported_channels:
  _type: beerocks::message::sWifiChannel
  _length: [ ]
```

---

:-1: dynamic length is very inconvenient because it is expressed in bytes, not in number of elements. So an explicit length is actually preferable.

(except if I'm mistaken and it really is expressed in number of elements)

---

Let me modify that statement: *unkown length* is annoying. *dynamic length* is good:
```suggestion
supported_channels_length: uint8_t
supported_channels:
  _type: "beerocks::message::sWifiChannel"
  _length: [ supported_channels_length ]
```

---

I wasn't aware, you are correct. IMO the length should return the number of elements and not the size of them, @LiorAmram what do you think?

---

Dynamic length means there is no explicit member on the buffer that indicates what is the length or the number of elements.
When building or parsing the class, there is an internal private size element that indicates the **number** of elements in the list. However, it can't be accessed from outside the class as far as I remember.
So dynamic length isn't expressed in bytes, it's actually isn't expressed at all.
When the discussion in whether to use dynamic length list (with length variable) or unknown length list (without length variable), I prefer using dynamic length lists whenever possible.
However, in this code, there is another issue: the specified length is a const, which means all elements of the list will be allocated in advance. Is it the desired output? If not, it might be preferable to define a length variable to supported_channels.

---

@arnout a small fix to your suggestion:
```suggestion
    _length: [ "beerocks::message::SUPPORTED_CHANNELS_LENGTH" ]  
supported_channels_length: 
  _type: uint8_t
  _length_var: True
supported_channels:
  _type: "beerocks::message::sWifiChannel"
  _length: [ supported_channels_length ]
```

Setting `_length_var: True` is necessary for tlvf to update this variable when allocating new elements to the list. Otherwise, it's just another member on the buffer.

---

The conversation is moot, the supported_channels was moved to the sNodeHostap